app/models/api_credential.rb

```diff
@@ -0,0 +1,78 @@
+require "digest"
+
+class ApiCredential < ApplicationRecord
+  belongs_to :user
+
+  before_save :generate_api_token
+  before_save :generate_refresh_token
+
+  # Securely confirm/deny that Hash in db is same as current users token Hash
+  def authenticate_api_token(api_token)
```

Comment:
does these functions return boolean value?
if so, the naming should be consistent with other similar functions starting with the is_....

Reply:
Yes, returns a boolean. How's Something like "is_api_token_valid"?

Reply:
yes that would be perfect
also follows ruby conventions too